Automattic/wp-calypso
# Hosting Command Palette: refactor the URL construction for wp-admin vs calypso links

Closes https://github.com/Automattic/dotcom-forge/issues/4764

## Proposed Changes

This PR refactors how the URL is constructed for the `wp-admin` vs. `Calypso` interface in the `SiteFunctions` in the command palette. More specifically, extracts the code below into a separate function to make the code less repetitive:

```
( site.jetpack && ! site.is_wpcom_atomic ) ||
						'wp-admin' === site.options?.wpcom_admin_interface
							? `${ site.URL }/wp-admin/plugin-install.php`
							: `/plugins/${ site.slug }`;
```

## Testing Instructions

* Navigate to Calypso Live Link
* Navigate to `/sites`
* Press `cmd + k` on macOS or `ctrl + k` on Windows to open a command palette
* Select a command that would navigate to a different UI based on the interface selected e.g. `Add new post`
* Make sure to select an Atomic site from nested commands
* Confirm that you are brought to either `Calypso` or `wp-admin` (depending which interface you have selected)
* Navigate to `https://wordpress.com/hosting-config/{yourAtomicSite}`
* Switch your interface to a different one
* Press `cmd + k` on macOS or `ctrl + k` on Windows to open a command palette
* Select the same command as previously selected
* Confirm that you are brought to the interface that you just selected
* Test multiple commands that were modified in this PR and confirm they behave as expected

## Pre-merge Checklist
